File: app/src/main/java/com/example/quote/QuoteSingleton.java
package com.example.quote;

import java.util.Collections;
import java.util.List;

class QuoteSingletonJava {
    private static List<Quote> quotes = Collections.emptyList();

    public static List<Quote> getQuotes() {
        return quotes;
    }

    public static void setQuotes(List<Quote> quotes) {
        QuoteSingletonJava.quotes = quotes;
    }
}

File: app/src/main/java/com/example/quote/QuoteSingleton.kt
package com.example.quote;

// Singleton object for managing quotes
object QuoteSingletonKotlin {
    private var quotes: List<Quote> = emptyList()

    fun getQuotes(): List<Quote> {
        return quotes
    }

    fun setQuotes(newQuotes: List<Quote>) {
        quotes = newQuotes
    }
}